### PR TITLE
fix creation failure

### LINKED ISSUE #7536: higher number of creation failures
https://cmsweb.cern.ch/reqmgr2/fetch?rid=fabozzi_Run2016B-v2-ZeroBias6-17Jan2017_8020_170117_170800_2500

https://cmsweb.cern.ch/reqmgr2/fetch?rid=fabozzi_Run2016B-v2-ZeroBias6-25Nov2016_8020_161125_152646_3953

have basically identical settings and yet, the 17Jan has had way more creationg failures. which is counter intuitive, has the code that generate that failure been changed much ?

### PR DESCRIPTION
**Pull request by ticoann**
fixes #7536

### CODE REVIEW

**Comment by ticoann** [comment]
@amaltaro, @vlimant, please review.

I think it is better to make configurable for jobTimeLimit, should it be request level configuration or agent level. If it only consider pilot limit, it can be agent level configuration but if we need fine control per request - not sure about the use case, it can be request level configuration.

**Comment by ticoann** [comment]
make it configurable in request level but haven't added the splitting parameter from web. It will only take the default value.

**Comment by ticoann** [comment]
@amaltaro, Alan could you review this patch?

File: src/python/WMCore/JobSplitting/EventAwareLumiBased.py [hunk 1]
```diff
@@ -35,6 +35,8 @@ def __init__(self, package='WMCore.DataStructs', subscription=None, generators=N
 
         self.loadRunLumi = None  # Placeholder for DAO factory if needed
         self.collectionName = None  # Placeholder for ACDC Collection Name, if needed
+        #TODO this might need to be configuarble instead of hardcoded
```
Comment (amaltaro) [inline src/python/WMCore/JobSplitting/EventAwareLumiBased.py]: typo

File: src/python/WMCore/JobSplitting/EventAwareLumiBased.py [hunk 2]
```diff
@@ -161,9 +163,9 @@ def algorithm(self, *args, **kwargs):
                 lumisInJobInFile = 0
                 updateSplitOnJobStop = False
                 failNextJob = False
-                # If the number of events per lumi is higher than the limit
+                # If estimated job time is higher the job time limt (condor limit)
```
Comment (amaltaro) [inline src/python/WMCore/JobSplitting/EventAwareLumiBased.py]: typo

File: src/python/WMCore/JobSplitting/EventAwareLumiBased.py [hunk 3]
```diff
@@ -161,9 +163,9 @@ def algorithm(self, *args, **kwargs):
...
                 # and it's only one lumi then ditch that lumi
-                if f['avgEvtsPerLumi'] > eventLimit and f['lumiCount'] == 1:
+                if(f['avgEvtsPerLumi'] * timePerEvent) > jobTimeLimit and f['lumiCount'] == 1:
```
Comment (amaltaro) [inline src/python/WMCore/JobSplitting/EventAwareLumiBased.py]: you missed a blank space in the beginning.
BTW, how about we add the job length as well to the error message on this line https://github.com/ticoann/WMCore/blob/ac9f6654cb85a69c37ba48f15302c6fc31792f3e/src/python/WMCore/JobSplitting/EventAwareLumiBased.py#L254

**Review by amaltaro — CHANGES_REQUESTED** [review]
Sorry for the delay, Seangchan.

Patch looks good in general, though I still think it will cause more problems to us when P&R team has to process workflows with very high TimePerEvent...

There are a few places that need to be updated as well (we need to be careful with backwards incompatibility here):
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ReqMgr/DataStructs/DefaultConfig/EDITABLE_SPLITTING_PARAM_CONFIG.py#L8

and also in a few StdSpecs

File: src/python/WMCore/JobSplitting/EventAwareLumiBased.py (hunk 3 above)
Comment (ticoann) [reply src/python/WMCore/JobSplitting/EventAwareLumiBased.py]: Do you want to print out total number of jobs?

**Comment by amaltaro** [comment]
How about we replace this message:
`"File %s has too many events (%d) in %d lumi(s)"`
by
`"File %s has a single lumi with too many events (%d) and it would take %s secs to run"`
?

**Comment by ticoann** [comment]
> "File %s has a single lumi with too many events (%d) and it would take %s secs to run"

secs? not hours? I thought you are relaxing Friday night. 😄

**Comment by vlimant** [comment]
what you have implemented is essentially making the max_events_per_lumi be 48h/timeperevent. Making the job_time_limit configurable is absolutely mandatory in this case. either we can set max_events_per_lumi or job_time_limit via script/interface.

I agree that in case of overestimated timeperevent, Ops would have to set job_time_limit to a very large value ; same as is done these days with setting max_events_per_lumi to an insanely large number.

@amaltaro isn't there a problem with avgEvtsPerLumi not beeing estimated properly when there is a lumi mask somewhere ? and hence leading to artificially large value of avgEvtsPerLumi, and these creation failures.

**Comment by ticoann** [comment]
> what you have implemented is essentially making the max_events_per_lumi be 48h/timeperevent. Making the job_time_limit configurable is absolutely mandatory in this case. either we can set max_events_per_lumi or job_time_limit via script/interface.

job_time_limit can be overwritten when splitting algorithm is set. The question is whether this is request property (for each request)  or general property over all the request. (It sounds like you want to make it request property, right?

> isn't there a problem with avgEvtsPerLumi not beeing estimated properly when there is a lumi mask somewhere ?

I don't think avgEvtsPerLumi will be related to lumi mask unless masked lumis has very different (small or large) event numbers then others. Mask will be applied later.

**Comment by amaltaro** [comment]
job_time_limit has to be a request property (actually step/task), just as TimePerEvent is.

About the lumi mask, no, there is no problem. Even if there was, the algorithm only fails job creation when the file has a single lumi section (where the avgEvtsPerLumi is precise on those cases)

**Comment by ticoann** [comment]
test this please

**Comment by amaltaro** [comment]
Seangchan, we need to change these parameters at ReqMgr level as well (and possible make it backwards compatible in the agent). I'll create another patch later today.

**Pull request merged**